torch/fx/experimental/symbolic_shapes.py

```diff
@@ -2108,15 +2223,36 @@ def __init__(
         self.fx_node_cache: Dict[Tuple[Callable, Tuple[Any, ...]], torch.fx.Node] = {}
         self.source_to_symbol: Dict[str, sympy.Symbol] = {}
 
+        # There can be only one event being recorded at once.
+        # Only the top-level method call is recorded.
+        self.recording_event: bool = not should_record_events
+
         if _translation_validation_enabled():
             from torch.fx.experimental.validator import TranslationValidator
 
+            # List of events for reconstructing ShapeEnv at arbitrary points in time.
+            self.events: List[ShapeEnvEvent] = [
+                ShapeEnvEvent(ShapeEnv, kwargs={
+                    "allow_scalar_outputs": allow_scalar_outputs,
+                    "allow_dynamic_output_shape_ops": allow_dynamic_output_shape_ops,
+                    "assume_static_by_default": assume_static_by_default,
+                    "specialize_zero_one": specialize_zero_one,
+                    "duck_shape": duck_shape,
+                    "frame_id": frame_id,
+                    "co_fields": co_fields,
+                    "should_record_events": False,
+                })
+            ]
+
+            self.tracked_fakes = tracked_fakes
```

Comment:
Note: this will almost certainly cause reference cycles, as tracked fakes has FakeTensor which has reference to SymInt which refer to ShapeEnv